src/basic/process-util.c

```diff
@@ -243,7 +243,15 @@ int get_process_cmdline(pid_t pid, size_t max_length, bool comm_fallback, char *
                 }
         }
 
-        *line = ans;
+        k = strdup(ans);
+        if (!k) {
+                free(ans);
+                return -ENOMEM;
+        }
+
+        free(ans);
```

Comment:
Please use `_cleanup_free_` instead. It'll be necessary to add `TAKE_PTR` in a few places.